### DISCUSSION TITLE
File.ExistsAsync and File.OpenAsync

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#74848] [Q&A] [answered]
I have a ton of files whose existence I need to check.  If they are on a really slow network share or on an SMB location that I don't have permission to access, it can take ~10-15 seconds to fail out. 

Is there an OpenAsync or ExistsAsync somewhere that I can call that I can cancel with either a token or a timeout?

**Comment by stephentoub** [comment, marked as answer]
There isn't. The OSes don't provide async versions of these operations.

**Reply by huoyaoyuan** [reply]
See also #67587 #63153

**Reply by TonyValenti** [reply]
Tell the OS folks to hop to it!

I can't tell you how many apps I've seen that are plagued by this.  Even Windows Explorer and especially the Start menu bug out pretty bad when there is a slow/inaccessible network path.
